### PR TITLE
Refactor MessageActions using ARIA Toolbar pattern 

### PR DESCRIPTION
**Pull request by jugglinmike**
This patch includes two commits to help reviewers distinguish between code reorganization and new implementation.

I haven't been able to test the alternate layout for "call in progress" because I don't have any contacts yet.

Signed-off-by: Mike Pennisi <mike@mikepennisi.com>

### CODE REVIEW

**Comment by t3chguy** [comment]
> This is intended to resolve vector-im/riot-web#11316

That issue is about the per-Message Actions Bar which has React, Reply, Edit, More Options
The buttons this changed were more like the Composer Actions. I'm not entirely sure what the best ARIA pattern for them is, toolbar doesn't seem insane but also keeping them as distinct buttons isn't terrible

**Comment by jugglinmike** [comment]
Got it. I've submitted gh-4827 to hopefully provide the desired fix. For this patch, is there anyone we should contact to help make a decision?

**Comment by t3chguy** [comment]
I would suggest chatting with the AT users in https://matrix.to/#/#a11y:riot.ovh as to what they'd find most obvious

**Comment by jugglinmike** [comment]
Oh, that's great! I'll follow up there

**Comment by jryans** [comment]
@jugglinmike Does https://github.com/matrix-org/matrix-react-sdk/pull/4827 replace this one, or are they both meant to be reviewed...?

**Comment by jugglinmike** [comment]
Hi @jryans! This patch is independent, so it's still fit for review. That said, since it wasn't what was requested in https://github.com/vector-im/riot-web/issues/11316, so @t3chguy and I have been talking with folks in the Matrix room mentioned above.

At least one person's on board, though to be honest, I don't know the criteria for moving forward with this patch.

**Comment by t3chguy** [comment]
Can you do a DCO sign-off in the PR description as one of your commits lacks one, thanks :)

File: src/components/views/rooms/MessageActions.js [hunk 1]
```diff
@@ -186,39 +194,103 @@ function VideoCallButton(props) {
...
-        <Stickerpicker key="sticker" room={props.room} />,
+        <UploadButton key="upload"
+                      tabIndex={tabIndex('upload')}
+                      roomId={props.room.roomId} />,
+        <EmojiButton key="emoji"
+                     tabIndex={tabIndex('emoji')}
+                     addEmoji={addEmoji} />,
+        <Stickerpicker key="sticker"
+                       tabIndex={tabIndex('sticker')}
+                       room={props.room} />,
```
Comment (t3chguy) [inline src/components/views/rooms/MessageActions.js:226]: our multi-line props style is to do 4 space indent on the line after the component opener

File: src/components/views/rooms/MessageActions.js [hunk 2]
```diff
@@ -186,39 +194,103 @@ function VideoCallButton(props) {
...
         } else {
             controls.push(
-                <CallButton key="call" roomId={props.room.roomId} />,
-                <VideoCallButton key="videocall" roomId={props.room.roomId} />,
+                <CallButton key="call"
+                            tabIndex={tabIndex('call')}
+                            roomId={props.room.roomId} />,
+                <VideoCallButton key="videocall"
+                                 tabIndex={tabIndex('videocall')}
+                                 roomId={props.room.roomId} />,
```
Comment (t3chguy) [inline src/components/views/rooms/MessageActions.js:243]: same as above

File: src/components/views/rooms/MessageActions.js [hunk 3]
```diff
@@ -186,39 +194,103 @@ function VideoCallButton(props) {
...
+                       room={props.room} />,
     ];
 
     if (props.showCallButtons) {
         if (callInProgress) {
             controls.push(
-                <HangupButton key="hangup" roomId={props.room.roomId} />,
+                <HangupButton key="hangup"
+                              tabIndex={tabIndex('hangup')}
+                              roomId={props.room.roomId} />,
```
Comment (t3chguy) [inline src/components/views/rooms/MessageActions.js:234]: same as above

**Comment by jugglinmike** [comment]
> Can you do a DCO sign-off in the PR description as one of your commits lacks one, thanks :)

Sure, although I don't intend for this branch to be merged as a series of distinct commits. Let me know if you would rather thatI squash them down on my end.

**Comment by jryans** [comment]
Thanks for making this contribution a while back. Since the code base has changed since this was opened, it no longer applies cleanly, and I don't think there's a need to keep it open in this state, as we can always find the code here again if needed. If you are still interested in pursuing this feature, please discuss with us in #element-dev:matrix.org to find a good approach forward.

**Pull request closed without merging**